src/sidebar/components/NotebookView.js

```diff
@@ -2,6 +2,7 @@ import { IconButton, Panel } from '@hypothesis/frontend-shared';
 import { useEffect, useLayoutEffect, useRef, useState } from 'preact/hooks';
 import scrollIntoView from 'scroll-into-view';
 
+import { PortFinder } from '../../annotator/communicator';
```

Comment:
I will do.

Are you happy to keep both classes (`FrameConnector` and `PortFinder`) on the same file? I thought because they mirror each other it is likely that you want to modify both. It could be tested in combination, too.